crates/core/src/client_events.rs

```diff
@@ -392,13 +392,16 @@ pub(crate) mod test {
                     }
                     val if (35..80).contains(&val) => {
                         if let Some(contract) = self.choose(&state.existing_contracts) {
-                            let delta = UpdateData::Delta(StateDelta::from(self.random_byte_vec()));
```

Comment:
This line should be moved to before the `choose` since the RNG state should be always the same regardless of any if conditions.

I don't think this was an issue for this case since all peers have the same `existing_contracts` at the same time but is dangerous to leave it like this.